mongo/integration/unified/matches.go

```diff
@@ -232,6 +232,22 @@ func evaluateSpecialComparison(ctx context.Context, assertionDoc bson.Raw, actua
 		if !bytes.Equal(expectedID, actualID) {
 			return fmt.Errorf("expected lsid %v, got %v", expectedID, actualID)
 		}
+	case "$$lte":
+		// Numeric values can be compared even if their types are different (e.g. if expected is an int32 and actual
+		// is an int64).
+		if assertionVal.IsNumber() {
+			if !actual.IsNumber() {
+				return fmt.Errorf("expected value to be a number but got a %s", actual.Type)
+			}
+
+			actualInt64 := actual.AsInt64()
+			expectedInt64 := assertionVal.AsInt64()
+			if actualInt64 > expectedInt64 {
+				return fmt.Errorf("expected numeric value %d to be less than or equal %d", actualInt64, expectedInt64)
+			}
+			return nil
+		}
+		return fmt.Errorf("expected assertionVal to be a number but got a %s", assertionVal.Type)
```

Comment:
```suggestion
		if !assertionVal.IsNumber() {
			return fmt.Errorf("expected assertionVal to be a number but got a %s", assertionVal.Type)
		}
		if !actual.IsNumber() {
			return fmt.Errorf("expected value to be a number but got a %s", actual.Type)
		}

		// Numeric values can be compared even if their types are different (e.g. if expected is an int32 and actual
		// is an int64).
		actualInt64 := actual.AsInt64()
		expectedInt64 := assertionVal.AsInt64()
		if actualInt64 > expectedInt64 {
			return fmt.Errorf("expected numeric value %d to be less than or equal %d", actualInt64, expectedInt64)
		}
		return nil
```
Completely optional to break up the [complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity).

Reply:
Yeah that reads much better; thanks. I also hadn't heard of cyclomatic complexity before 🤔